modules/olmv1-creating-a-cluster-role.adoc

```diff
@@ -0,0 +1,334 @@
+
+. Search for the `clusterrole` and `clusterrolebindings` values in the `rules.resources` field in the extension's CSV file. Copy the API groups, resources, verbs, and resource names to your manifest, similar to the following example:
++
+.Example cluster role manifest
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pipelines-installer-clusterrole
+rules:
+# ...
+# ClusterRoles and ClusterRoleBindings for the controllers of the extension
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  verbs:
+  - create # <1>
+  - list
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  verbs:
+  - get
+  - update
+  - patch
+  - delete
+  resourceNames: # <2>
+  - "*"
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterrolebindings
+  verbs:
+  - create
+  - list
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterrolebindings
+  verbs:
+  - get
+  - update
+  - patch
+  - delete
+  resourceNames:
+  - "*"
+# ...
+----
+<1> You cannot scope `create`, `list`, and `watch` permissions to specific resource names (the `resourceNames` field). You must scope these permissions to their resources (the `resources` field).
+<2> Some resource names are generated by using the following format: `<package_name>.<hash>`. After you install the extension, look up the resource names for the cluster roles and cluster role bindings for the controller of the extension. Replace the wildcard characters in this example with the generated names and follow the principle of least privilege.
+
+. Search for the `customresourcesdefinition` value in the `rules.resources` field in the extension's CSV file. Copy the API groups, resources, verbs, and resource names to your manifest, similar to the following example:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pipelines-installer-clusterrole
+rules:
+# ...
+# Custom resource definitions of the extension
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - create
+  - list
+  - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - update
+  - patch
+  - delete
+  resourceNames:
+  - manualapprovalgates.operator.tekton.dev
+  - openshiftpipelinesascodes.operator.tekton.dev
+  - tektonaddons.operator.tekton.dev
+  - tektonchains.operator.tekton.dev
+  - tektonconfigs.operator.tekton.dev
+  - tektonhubs.operator.tekton.dev
+  - tektoninstallersets.operator.tekton.dev
+  - tektonpipelines.operator.tekton.dev
+  - tektonresults.operator.tekton.dev
+  - tektontriggers.operator.tekton.dev
+# ...
+----
+
+. Search the CSV file for stanzas with the `permissions` and `clusterPermissions` values in the `rules.resources` spec. Copy the API groups, resources, verbs, and resource names to your manifest.
```

Comment:
```suggestion
. Search the CSV file for stanzas with the `permissions` and `clusterPermissions` values in the `rules.resources` specification. Copy the API groups, resources, verbs, and resource names to your manifest.
```

Reply:
I don't think I've seen .spec used as a noun like that before. You could do the `.spec.rules.resources` field/value.

Reply:
"Spec" is in the repo glossary: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/term_glossary.adoc#spec